packages/rum-recorder/src/domain/record/serializationUtils.ts

```diff
@@ -38,101 +39,45 @@ export function nodeOrAncestorsIsIgnored(n: NodeWithSerializedNode) {
 }
 
 export function transformAttribute(doc: Document, name: string, value: string): string {
-  // relative path in attribute
-  if (name === 'src' || (name === 'href' && value)) {
-    return makeUrlAbsolute(doc, value)
-  }
-  if (name === 'srcset' && value) {
-    return makeSrcsetUrlsAbsolute(doc, value)
-  }
-  if (name === 'style' && value) {
-    return makeStylesheetUrlsAbsolute(value, location.href)
+  if (value) {
+    if (name === 'src' || name === 'href') {
+      return makeUrlAbsolute(value, doc.location.href)
+    }
+    if (name === 'srcset') {
+      return makeSrcsetUrlsAbsolute(value, doc.location.href)
+    }
+    if (name === 'style') {
+      return makeStylesheetUrlsAbsolute(value, doc.location.href)
+    }
   }
   return value
 }
 
 const URL_IN_CSS_REF = /url\((?:(')([^']*)'|(")([^"]*)"|([^)]*))\)/gm
-const RELATIVE_PATH = /^(?!www\.|(?:http|ftp)s?:\/\/|[A-Za-z]:\\|\/\/).*/
-const DATA_URI = /^(data:)([^,]*),(.*)/i
-export function makeStylesheetUrlsAbsolute(cssText: string | null, href: string): string {
-  return (cssText || '').replace(
+const ABSOLUTE_URL = /^[A-Za-z]:|^\/\//
+const DATA_URI = /^data:.*,/i
```

Comment:
For DATA_URI: the new regex in functionaly the same as the old one (`/^(data:)([^,]*),(.*)/i`): it matches a string starting with `data:`, with a comma later on. The capturing parenthesis are not used.

For ABSOLUTE_URL: the old regex to match strings starting with `www.`, `http://` and associated, single letter drive Windows absolute path (ex: `C:\foo`), and protocol-less URLs `//`.

* The first case (`www.`) was incorrect: URLs starting with `www.` are not absolute:
```js
new URL('www.foo.com', 'http://example.org').href
// "http://example.org/www.foo.com"
```
* By testing, I figured out that any URL starting with characters followed by a semicolon was considered as an absolute URL, so `http:` and `C:` checks can be factorized.
* The last case, protocol-less URLs, has been kept.

Please note that those regexes are not really necessary since `makeUrlAbsolute` handles absolute URLs and data URLs correctly. I think they could be kept to avoid parsing many absolute or DATA URLs unnecessarily (that could be a performance bottleneck).